### PR TITLE
[local companion app] Add to dashboard

### PR DESCRIPTION
**Pull request by csweichel**
This PR makes the local companion app available from the dashboard, i.e. adds it to the `public/` assets.

It also sets the version during build by overwriting a `version.go` file instead of using ldflags. This way we can set the ldflags globally in the `WORKSPACE.yaml`.

### How to test
```bash
# Download the local companion app binary from the dashboard
curl -OL https://cw-lca-dashboard-dl.staging.gitpod-dev.com/static/bin/gitpod-local-companion-linux

# Run the app and check its version
chmod +x gitpod-local-companion-linux
./gitpod-local-companion-linux -v
```

### CODE REVIEW

File: WORKSPACE.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ defaultArgs:
   coreYarnLockBase: ../..
   npmPublishTrigger: "false"
   publishToNPM: true
+  localAppVersion: unkown
```
Comment (rl-gitpod) [inline WORKSPACE.yaml:7]: /minor nit
-> unknown

File: chart/templates/proxy-deployment.yaml [hunk 2]
```diff
@@ -39,13 +39,15 @@ metadata:
     component: proxy
     kind: deployment
     stage: {{ .Values.installation.stage }}
+    gitpod.io/requiresCerts: 'true'
```
Comment (rl-gitpod) [inline chart/templates/proxy-deployment.yaml:42]: Some bonus, but important additions :-)

File: components/local-app/BUILD.yaml [hunk 3]
```diff
@@ -42,13 +48,26 @@ packages:
...
+        - ["cp", "_deps/components-local-app--version/version.go", "version.go"]
       config:
         packaging: app
+    - name: version
+      type: generic
+      argdeps:
+        - localAppVersion
+      config:
+        commands:
+          - ["sh", "-c", "echo package main >> version.go"]
```
Comment (rl-gitpod) [inline components/local-app/BUILD.yaml:67]: fyi - the embed directive in Golang v1.16 can also be used for this e.g. https://www.buymeacoffee.com/smarterfish/dynamic-version-numbers-in-go-1-16

It has the bonus of not requiring a valid .go file and also standardises the way to embed any sort of static file

**Review by rl-gitpod — APPROVED** [review]

File: components/local-app/BUILD.yaml (hunk 3 above)
Comment (csweichel) [reply components/local-app/BUILD.yaml:67]: Excellent point. We're using `embed` in `docker-up` already, might as well use it here.

**Pull request merged**